include/lokinet/lokinet_udp.h

```diff
@@ -2,103 +2,119 @@
 
 #include "lokinet_context.h"
 
-#ifdef _WIN32
-extern "C"
-{
-  struct iovec
-  {
-    void* iov_base;
-    size_t iov_len;
-  };
-}
-#else
-#include <sys/uio.h>
-#endif
-
 #ifdef __cplusplus
 extern "C"
 {
 #endif
 
   /// information about a udp flow
-  struct lokinet_udp_flow
+  struct lokinet_udp_flowinfo
   {
-    /// the socket id for this flow used for i/o purposes and closing this socket
-    int socket_id;
     /// remote endpoint's .loki or .snode address
-    char remote_addr[256];
-    /// local endpoint's ip address
-    char local_addr[64];
+    char remote_host[256];
     /// remote endpont's port
-    int remote_port;
-    /// local endpoint's port
-    int local_port;
+    uint16_t remote_port;
+    /// the socket id for this flow used for i/o purposes and closing this socket
+    int socket_id;
   };
 
-  /// establish an outbound udp flow
-  /// remoteHost is the remote .loki or .snode address conneting to
-  /// remotePort is either a string integer or an srv record name to lookup, e.g. thingservice in
-  /// which we do a srv lookup for _udp.thingservice.remotehost.tld and use the "best" port provided
-  /// localAddr is the local ip:port to bind our socket to, if localAddr is NULL then
-  /// lokinet_udp_sendmmsg MUST be used to send packets return 0 on success return nonzero on fail,
-  /// containing an errno value
-  int EXPORT
-  lokinet_udp_establish(
-      char* remoteHost,
-      char* remotePort,
-      char* localAddr,
-      struct lokinet_udp_flow* flow,
-      struct lokinet_context* ctx);
-
   /// a result from a lokinet_udp_bind call
   struct lokinet_udp_bind_result
   {
     /// a socket id used to close a lokinet udp socket
     int socket_id;
   };
 
+  /// flow acceptor hook, return 0 success, return nonzero with errno on failure
+  typedef int (*lokinet_udp_flow_filter)(
+      void* /*user*/,
+      const struct lokinet_udp_flowinfo* /* remote address */,
+      void** /* flow-userdata */,
+      int* /* timeout seconds */);
```

Comment:
You can have argument names in a function typedef, so these can all get uncommented.

Reply:
names added